ambari-server/src/main/java/org/apache/ambari/server/state/ServiceGroupImpl.java

```diff
@@ -135,6 +140,20 @@ public long getClusterId() {
     return cluster.getClusterId();
   }
 
+  @Override
+  public String getStackId() {
+    ServiceGroupEntity entity = getServiceGroupEntity();
+    return entity.getStackId();
+  }
+
+  @Override
+  public void setStackId(String stackId) {
+    ServiceGroupEntity entity = getServiceGroupEntity();
+    entity.setStackId(stackId);
```

Comment:
I don't understand why this is a string and not an mpack association. The mpack is what will give us information about the stack, version, etc.